kartothek/io/dask/dataframe.py

```diff
@@ -73,9 +78,18 @@ def read_dataset_as_ddf(
         label_filter=label_filter,
         dates_as_object=dates_as_object,
         predicates=predicates,
+        dispatch_by=dask_index_on,
     )
-
-    return dd.from_delayed(delayed_partitions, meta=meta)
+    if dask_index_on:
+        divisions = ds_factory.indices[dask_index_on].observed_values()
+        divisions.sort()
+        divisions = list(divisions)
+        divisions.append(divisions[-1])
+        return dd.from_delayed(
+            delayed_partitions, meta=meta, divisions=divisions
+        ).set_index(dask_index_on, divisions=divisions)
```

Comment:
I believe I will need to add the sorted kwarg here to avoid a shuffle

- [ ] Add sorted kwarg and try to figure out if we can test for shuffle